src/Microsoft.Graph/Constants.cs

```diff
@@ -77,5 +77,52 @@ public static class OdataInstanceAnnotations
             /// </summary>
             public const string DeltaLink = "@odata.deltaLink";
         }
+
+        /// <summary>
+        /// Constants used to specify content types
+        /// </summary>
+        public static class ContentTypes
+        {
+            /// <summary>
+            /// The deltaLink annotations string.
+            /// </summary>
+            public const string JsonContentType = "application/json";
+
+            /// <summary>
+            /// The deltaLink annotations string.
+            /// </summary>
+            public const string StreamContentType = "application/octet-stream";
+        }
+
+        /// <summary>
+        /// Constants used to specify Http methods
+        /// </summary>
+        public static class HttpMethods
```

Comment:
Would using enum or something like https://stackoverflow.com/questions/630803/associating-enums-with-strings-in-c-sharp be more efficient in  place of HttpMethods?

Reply:
I think enums hold integers. Therefore we would need to add code that will translate the integers to strings. Accessing them like this is more efficient as the methods are strings.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.enum.getname?view=net-5.0

Reply:
Thanks @baywet. 
I still think it would be better to access them as is. 
Otherwise would replace the strings with something like this?

```cs
this.Method = Enum.GetName(typeof(HttpMethods), 3);
```

Reply:
I think @nikithauc 's suggestion was to use an enum instead of a string for all the service library all the way down to the core lib. And in the core lib the parsing/serialization of the enum value would happen. As of this method you can perfectcly write
```CSharp
Enum.GetName(typeof(HttpMethods), HttpMethods.Get)
```

Reply:
https://stackoverflow.com/questions/613837/enums-and-constants-which-to-use-when 
Enum makes sense in this case. Please refer to the above link.

My suggestion is to use an Enum HttpMethods if possible in place of the constants  class as introduced in this. An example of this is https://stackoverflow.com/questions/630803/associating-enums-with-strings-in-c-sharp.

Reply:
If the enum values match the string we want, we don't need to set string values :)

Reply:
I think having string constants is not a problem if most usages are strings. It may be a little less performant if we are using them in comparisons because it will be doing a string comparison instead of int.

If you don't like quoted strings here is one way to avoid it:
https://github.com/microsoftgraph/msgraph-sdk-raptor/blob/2c7252312b45468b266ec33476dec02da88f4bcc/TestsCommon/TestDataGenerator.cs#L69-L73
But note that some linters may complain on naming convention with all caps.

All this said, these constants except for PATCH exist in `System.Net.Http.HttpMethod` where `Method` returns string representation (e.g. `System.Net.Http.HttpMethod.Delete.Method == "DELETE"`)

If, someday, we change the minimum dotnet framework we are targeting, we can get patch too: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmethod.patch?view=netcore-2.2#System_Net_Http_HttpMethod_Patch]

Reply:
Thanks guys. 
Since there exists a relationship between the constants I now see that there exists value in holding the values in an enum. Will therefore action and resolve asap.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.enum.tostring?view=net-5.0

Could also work out.

Reply:
Resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/905/commits/5e56de9fba1fbf07ee4163e426d5e354bfc79747

Reply:
@zengin 
For consistency(due to the missing Patch) I have currently not used the System.Net.Http values.
When we get around to upgrading to netstandard2.1 it will just be a simple matter of updating the values here appropriately in the Generator and we should be good to go.

https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/blob/feature/dotnet-v2/src/GraphODataTemplateWriter/CodeHelpers/CSharp/CodeWriterCSharp.cs#L31

Reply:
sorry, but there's still something I'm not clear about, why is the method property of type string instead of enum? The enum should be defined in the core library, and the value carried over all the way to the core library. This would make comparisons faster in conditions as @zengin mentioned earlier. It'd would also reduce the risk of getting a unexpected value.

Reply:
@baywet. The property is actually of type enum in latest update.
But now I see the point on moving to the enum to the core library. I shall resolve this in the next update.

Reply:
Core PR update https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/905

Reply:
This is now resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/905/commits/4824f2c9bca235525a3eac24bfb8f0092836cb48 and https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/216

Reply:
maybe I didn't express myself clearly.
here https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/905/files#diff-e513f50737ffecefb6cf1ede5e9b115672f6ea707336933953cc55cc36bc2ce7R47 why do we have `this.Method = CoreConstants.HttpMethods.POST.ToString()` instead of `this.Method = CoreConstants.HttpMethods.POST` ? (having the method property as an enum and not a string)

Reply:
@baywet We are setting this property which is a string
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/c914abb65a3c0e388166fd26f21d187325180da2/src/Microsoft.Graph.Core/Requests/BaseRequest.cs#L87
Which is then used to create a [HttpRequestMessage ](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.-ctor?view=net-5.0#System_Net_Http_HttpRequestMessage__ctor_System_Net_Http_HttpMethod_System_String_)to send out which takes a string in its constructor. See [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/c914abb65a3c0e388166fd26f21d187325180da2/src/Microsoft.Graph.Core/Requests/BaseRequest.cs#L275)

Reply:
Why not having that property as an enum instead? You could then call the ToString() once (last link in your previous comment) instead of thousands of times.

Reply:
I believe that would break/change the [IBaseRequest](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/dev/src/Microsoft.Graph.Core/Requests/IBaseRequest.cs#L33) interface and all dependent code(as well as customers' code) that assumes the Method property to be a string. We would furthermore be saying to customers that if a method is not in the list of enums you can't use it when making modifying requests(assuming you would want to).

I think if we were to change the type the best way would be to use the [HttpMethod](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmethod?view=net-5.0) class provided by .NET as Mustafa suggested but the current runtime we are using is missing some of the methods we need(i.e Patch).

If we really want to break customers in this way, I would be happy to change this in a separate PR(once we all agree that breaking IBaseRequest is something we really want to do)

Reply:
thanks for the additional information @andrueastman . There's something I'm not clear about: are we not planning to major bump core as well as we release the new system.text.json implementation? In terms of "blocking the customers from using other HTTP methods" I don't believe that'd be a problem if we added all the existing HTTP methods (even the ones we're not using) in the enum. https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods

Reply:
Adding all the methods to the enum makes sense to me.
I shall resolve and update this in the next update.

Reply:
thanks for getting to the bottom of this discussion :)

Reply:
Resolved via https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/905/commits/3c46cb40e8c3f9be4c2d2a4360ee34d002357d6a